### PR TITLE
Improving the versatility of the HTTP output binding

### PR DESCRIPTION
**Pull request by pkedy**
# Description

Improving the versatility of the HTTP output binding. Removed the `Read` method since it is not an input binding.

## Issue reference

Related to dapr/docs#153. Will update the docs for this binding as a follow up.

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#153

### CODE REVIEW

**Comment by pkedy** [comment]
Please review @yaron2 @artursouza

**Review by mukundansundar — COMMENTED** [review]
@pkedy If Read is being removed ... please change to conformance test config too ....

**Review by mukundansundar — COMMENTED** [review]
@pkedy 
This seems like a breaking change ... it removes HTTP as an input binding.  cc @yaron2 @artursouza 

This also breaks the current conformance tests defined for HTTP binding.

File: bindings/http/http.go [hunk 1]
```diff
@@ -36,69 +43,126 @@ func NewHTTP(logger logger.Logger) *HTTPSource {
...
+		body = bytes.NewBuffer(req.Data)
+	case "GET", "HEAD", "DELETE", "OPTIONS", "TRACE":
+	default:
+		return nil, fmt.Errorf("invalid operation: %s", req.Operation)
 	}
 
-	return b, nil
-}
-
-func (h *HTTPSource) Read(handler func(*bindings.ReadResponse) error) error {
```
Comment (artursouza) [inline bindings/http/http.go:75]: Why removing Read?

**Review by artursouza — CHANGES_REQUESTED** [review]

**Comment by yaron2** [comment]
> @pkedy
> This seems like a breaking change ... it removes HTTP as an input binding. cc @yaron2 @artursouza
> 
> This also breaks the current conformance tests defined for HTTP binding.

@mukundansundar this actually fixes the HTTP binding as it was never meant to be an input binding. the test must change accordingly. /cc @pkedy

**Comment by artursouza** [comment]
> > @pkedy
> > This seems like a breaking change ... it removes HTTP as an input binding. cc @yaron2 @artursouza
> > This also breaks the current conformance tests defined for HTTP binding.
> 
> @mukundansundar this actually fixes the HTTP binding as it was never meant to be an input binding. the test must change accordingly. /cc @pkedy

Why not?

**Comment by yaron2** [comment]
> > > @pkedy
> > > This seems like a breaking change ... it removes HTTP as an input binding. cc @yaron2 @artursouza
> > > This also breaks the current conformance tests defined for HTTP binding.
> > 
> > 
> > @mukundansundar this actually fixes the HTTP binding as it was never meant to be an input binding. the test must change accordingly. /cc @pkedy
> 
> Why not?

In the older implementation, it was an input binding, and input bindings are blocking as they continuously get events. So for the HTTP binding, in practice this meant that the URL defined in the Component was only invoked *once* during startup and never again.

The HTTP binding was really meant to be a bi-directional binding where a user can invoke an external system over HTTP and get the data back.

File: bindings/http/http.go (hunk 1 above)
Comment (yaron2) [reply bindings/http/http.go:75]: See here: https://github.com/dapr/components-contrib/pull/661#issuecomment-775525544

**Comment by artursouza** [comment]
It can be input binding where it gets dat

> > > > @pkedy
> > > > This seems like a breaking change ... it removes HTTP as an input binding. cc @yaron2 @artursouza
> > > > This also breaks the current conformance tests defined for HTTP binding.
> > > 
> > > 
> > > @mukundansundar this actually fixes the HTTP binding as it was never meant to be an input binding. the test must change accordingly. /cc @pkedy
> > 
> > 
> > Why not?
> 
> In the older implementation, it was an input binding, and input bindings are blocking as they continuously get events. So for the HTTP binding, in practice this meant that the URL defined in the Component was only invoked _once_ during startup and never again.
> 
> The HTTP binding was really meant to be a bi-directional binding where a user can invoke an external system over HTTP and get the data back.

The input binding could trigger an infinite loop getting from the URL every second or so. I am fine removing it for now.

**Review by artursouza — APPROVED** [review]

**Pull request merged**